stdlib/asm/crypto/stark/verifier.masm

```diff
@@ -0,0 +1,1439 @@
+
+    repeat.235
+        exec.random_coin::generate_next_extfelt_tuple
+        loc_loadw.0
+        # => [Seed, 2, ptr + 1]
+    end
+    # => [Seed, x, x]
+
+
+    # Reseed with constraint composition polynomial commitment
+    swapw
+    adv_loadw
+    loc_storew.4
+    swapw
+    exec.random_coin::reseed
+    loc_storew.0
+    #=> [Seed_n, X, Y, ]
+
+
+    # Draw OOD z
+    push.1.0.0.0
+    loc_storew.20
+    swapw
+    push.1.1.0.0
+    hperm
+
+    # TODO: optimize this
+    dropw
+    swapw
+    dropw 
+    swapw
+    dropw
+
+    # Store OOD challenge z and z^8 so that when it is loaded on stack as [z^8_1, z^8_0, z1, z0]
+    drop drop
+    dup.1
+    dup.1
+    repeat.3
+        dup.1
+        dup.1
+        ext2mul
+    end
+    loc_storew.2251
+    # => [X, Y, x, x, 0, 0, 0, 0, 0, 0]
+
+
+    # Read [ood_main_trace_frame.current(), aux_trace_frame.current()]
+    # We have 72 main trace columns and 9 aux trace columns for a total of 162 base field elements
+    # 162 = 20 * 8 + 2
+    # [ood_main_trace_frame.current(), aux_trace_frame.current()] is stored in locaddr.2252 to locaddr.2292.
+    # They are stored from the stack as (a1_1, a1_0, a0_1, a0_0) except for locaddr.2292
+    # which is stored from the stack as (a1, a0, 1, 0)
+    locaddr.2252
+    movdn.8
+    push.1.0.0.0
+    swapw.2
+    repeat.20
+        adv_pipe
+    end
+
+    # adv_pipe the last two elements
+    dropw dropw
+    adv_push.1
+    adv_push.1
+    push.1
+    push.0
+    loc_storew.2292
+    push.0.0.0.0
+    loc_storew.21
+    hperm
+    # => [X, Digest_current, X, ptr]
+
+    # Reseed
+    loc_loadw.0
+    exec.random_coin::reseed
+    loc_storew.0
+    # => [Seed_n, X, X, X, ptr]
+    
+    # TODO: optimize this
+    dropw dropw drop drop drop
+    # => [X, X, ..]
+
+    # Read [ood_main_trace_frame.next(), aux_trace_frame.next()]
+    # We have 72 main trace columns and 9 aux trace columns for a total of 162 base field elements
+    # 162 = 20 * 8 + 2
+    # [ood_main_trace_frame.next(), aux_trace_frame.next()] is stored in locaddr.2293 to locaddr.2333.
+    # They are stored from the stack as (a1_1, a1_0, a0_1, a0_0) except for locaddr.2333
+    # which is stored from the stack as (a1, a0, 1, 0)
+    locaddr.2293
+    movdn.8
+    push.1.0.0.0
+    swapw.2
+    repeat.20
+        adv_pipe
+    end
+
+    # adv_pipe the last two elements
+    dropw dropw
+    adv_push.1
+    adv_push.1
+    push.1
+    push.0
+    loc_storew.2333
+    push.0.0.0.0
+    hperm
+    # => [X, Digest_current, X, ptr]
+
+    # Reseed
+    loc_loadw.0
+    exec.random_coin::reseed
+    loc_storew.0
+    # => [Seed_n, X, X, X, ptr]
+
+    # Read OOD constraint evaluations. These are 8 ExtFelt `value_i` such that the value of
+    # constraint evaluation function at `z` `H(z)` equals `\sum_0^7 z^i value_i`.
+    # The values are stored from locaddr.2334 to locaddr.2337
+    locaddr.2334
+    movdn.8
+    push.0.0.0.0
+    swapw.2
+    adv_pipe
+    adv_pipe
```

Comment:
I'd put this into a stand-alone procedure - e.g., `load_ood_constraint_evaluations`.